modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -0,0 +1,204 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayMultiSelect from '@clayui/multi-select';
+import classNames from 'classnames';
+import {fetch} from 'frontend-js-web';
+import React, {useContext, useEffect, useRef, useState} from 'react';
+
+import {addToCart} from '../add_to_cart/data';
+import MiniCartContext from './MiniCartContext';
+
+const CHANNEL_RESOURCE_ENDPOINT =
+	'/o/headless-commerce-delivery-catalog/v1.0/channels';
+
+const channelsApiURL = new URL(
+	`${themeDisplay.getPathContext()}${CHANNEL_RESOURCE_ENDPOINT}`,
+	themeDisplay.getPortalURL()
+);
+
+const formattedProducts = [];
+
+const ProductAutocompleteList = ({onItemClick, sourceItems}) => (
+	<ClayDropDown.ItemList>
+		{sourceItems.map((product) => {
+			const {id, label, value} = product;
+
+			return (
+				<ClayDropDown.Item
+					key={id}
+					onClick={() => onItemClick(product)}
+				>
+					<div className="autofit-row autofit-row-center">
+						<div className="autofit-col mr-3 w-25">{value}</div>
+
+						<span className="ml-2 text-truncate-inline">
+							<span className="text-truncate">{label}</span>
+						</span>
+					</div>
+				</ClayDropDown.Item>
+			);
+		})}
+	</ClayDropDown.ItemList>
+);
+
+export default function CartQuickAdd() {
+	const {cartState, setCartState} = useContext(MiniCartContext);
+
+	const [productsQuery, setProductsQuery] = useState('');
+	const [quickAddToCartError, setQuickAddToCartError] = useState(false);
+	const [selectedProducts, setSelectedProducts] = useState([]);
+	const [products, setProducts] = useState();
+
+	const channelIdRef = useRef();
+
+	const {cartItems = []} = cartState;
+
+	useEffect(() => {
+		async function fetchProducts() {
+			const channelsResponse = await fetch(channelsApiURL.toString());
+
+			const channels = await channelsResponse.json();
+
+			channelIdRef.current = channels.items[0].id;
+
+			const productsApiURL = new URL(
+				`${themeDisplay.getPathContext()}${CHANNEL_RESOURCE_ENDPOINT}/${
+					channelIdRef.current
+				}/products?nestedFields=skus`,
+				themeDisplay.getPortalURL()
+			);
+
+			const productsResponse = await fetch(productsApiURL.toString());
+
+			const availableProducts = await productsResponse.json();
+
+			availableProducts.items.forEach((product) => {
+				const {name, skus} = product;
+
+				formattedProducts.push({
+					...product,
+					label: name,
+					value: skus[0].sku,
+				});
+			});
+
+			setProducts(availableProducts.items);
+		}
+
+		fetchProducts();
+	}, []);
+
+	const handleAddToCartClick = () => {
+		const itemSKUs = selectedProducts.map((item) => item.value);
+		const readyProducts = [];
+
+		products.forEach((product) => {
+			if (itemSKUs.includes(product.skus[0].sku)) {
```

Comment:
Can `product.skus[0]` ever be null?

Reply:
I do not have access to that specific information, from what I can see in the API, it is never null as its a defining factor of a product. Like a barcode represented by a string.

Reply:
Since we don't necessarily know if product.skus[0] could be null, or if product.skus could be an empty array, we need to make sure and guard this check.

```suggestion
			if (product.skus.length && product.skus[0] && itemSKUs.includes(product.skus[0].sku)) {
```